### ISSUE TITLE
A new minimal profile

### ISSUE THREAD

**Issue by kilasuit** [PowerShell/PowerShell#18670]
### Summary of the new feature / enhancement

As a user that is troubleshooting issues I may want a very minimal profile that isn't my full profile, which may set things like host ui settings, prompt and colours to differentiate between sessions.

### Proposed technical implementation details (optional)

This should be able to be called via a new -minprofile switch on the pwsh executable & be set like any other profile but only ever called when that switch is present

**Comment by kilasuit** [comment]
This would be an extension to the current noprofile option and could be used in conjunction with it
This also would be great to be added as an additional jumplist entry for those of us on windows as part of install, like I have previously requested.

**Comment by SteveL-MSFT** [comment]
Instead of `-minprofile`, perhaps it would make sense to have `-profile <path>`?

**Comment by mklement0** [comment]
I like the idea, @SteveL-MSFT; it would be (loosely) analogous to `bash --rcfile <path>`. 

The semantics would have to be clarified either way, though: is the specified file then the _only_ profile that gets loaded,  or does it merely override `$PROFILE.currentUserCurrentHost`, meaning that all other profiles (if present) are loaded too, as usual?

**Comment by SteveL-MSFT** [comment]
@mklement0 good point, my thinking is that it's an override of `$PROFILE.CurrentUserCurrentHost`

**Comment by kilasuit** [comment]
> Instead of `-minprofile`, perhaps it would make sense to have `-profile <path>`?

@SteveL-MSFT  - Personally I prefer to look at things with `As well as` in mind over `instead of's` because to me that feels more inclusive and may be overall easier on end users and usecases, though there are definitely times where an `Instead of` is required, but I don't think that is the case here.
So for me I think that's certainly an option to have both, though if we were to keep things as close to the current profile mechanisms that are in place (at least for now) then just adding additional `Minimal` options would IMO be the simplest way forward for now & would allow us to easily then expand on this by other means in future like how you've suggested.

Though looking into HostUtilities.cs it seems there may have even been an option for this that we could potentially more easily expose, that is If I have read the code right though I may not have done so am going away and re-reading it & thinking on it.

**Comment by rkeithhill** [comment]
WG-Engine discussed this today.

Since you can get the desired functionality today by using `pwsh -noe -c '. <path-to-minprofile.ps1>'`, we do not see a strong enough case for adding this feature.

**Comment by kilasuit** [comment]
Whilst I can see where the Engine WG are coming from, because that is `relatively` easy, I feel that for end users/administrators/developers/hobbists etc they should be able to define this in a way that is just as simple as it is for how we currently allocate the already available profiles. After all, PowerShell is all about optimising the users time and experience where best possible.

Now this may end up being things that as a wider discussion turns into the things needed in a minimal profile that will help in troubleshooting issues in this repo or in scripts/modules etc going forward. 
I think this needs a verbal conversation across at least the Working Groups, but may be worthwhile being a point of topic for a future community call for wider input.

**Comment by kilasuit** [comment]
@mklement0 @rkeithhill @SteveL-MSFT -  Just as an FYI I ended up building this recently (yanno to cure the itch) and have it available [in this branch](https://github.com/kilasuit/PowerShell/tree/pwsh-executable-minprofile-parameter) 

It automatically implements a `-nop -noe -c ` & allows calling via `pwsh -minprofile  C:\Code\Mine\pub\GitHub\kilasuit\PSProfile\src\min-profile.ps1` 

Wouldn't be much to change it from minprofile just to profile either & gives this output

<img width="772" height="165" alt="Image" src="https://github.com/user-attachments/assets/edc01185-db7c-4477-9122-bf697d6fe533" />
